docs/preview/features/correlation.md

```diff
@@ -108,4 +108,64 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 }
 ```
 
+## Using secret store within Azure Functions
+
+### Installation
+
+For this feature, the following package needs to be installed:
+
+```shell
+PM > Install-Package Arcus.WebApi.Logging.AzureFunctions
+```
+
+### Usage
+
+To make sure the correlation is added to the HTTP response, following additions have to be made in the `.Configure` methods of the function's startup:
+
+```csharp
+[assembly: FunctionsStartup(typeof(Startup))]
+
+namespace MyHttpAzureFunction
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.AddHttpCorrelation();
+        }
+    }
+}
+```
+
+When this addition is added, you can use the `HttpCorrelationService` inside your function to call the correlation functionality and use the `ICorrelationInfoAccessor` (like before) to have access to the `CorrelationInfo` of the HTTP request.
+
+```csharp
+public class MyHttpFunction
+{
+    private readonly HttpCorrelationService _correlationService;
+    private readonly ICorrelationInfoAccessor _correlationAccessor;
+
+    public MyHttpFunction(HttpCorrelationService correlationService, ICorrelationInfoAccessor correlationAccessor)
+    {
+        _correlationService = correlationService;
+        _correlationAccessor = correlationAccessor;
```

Comment:
What if you could access the correlation accessor through the HttpCorrelation object?

Reply:
Yes, that was something that I also suggested [here](https://github.com/arcus-azure/arcus.webapi/pull/171#discussion_r453570179). It could be added value since you then only have to inject 1 correlation model.

Reply:
Let's do it!

Reply:
I'll added it as a property, but it got me wondering if we should let the `HttpCorrelation` implement the `ICorrelationInfoAccessor` interface and delegate it to (now private) field.

Reply:
I would let `HttpCorrelation` implement `ICorrelationInfoAccessor` so that I can use that as my single place for HTTP correlation.


> it got me wondering if we should let the `HttpCorrelation` implement the `ICorrelationInfoAccessor` interface and delegate it to (now private) field.

If you are talking about delegating the fields, why not simply use `HttpCorrelationInfoAccessor` inside `HttpCorrelation` instead of implementing it or am I misunderstanding your suggestion?

Reply:
> I would let HttpCorrelation implement ICorrelationInfoAccessor so that I can use that as my single place for HTTP correlation.

That's actually my suggestion. 😄 . So, yes, some misunderstanding here.

Reply:
I like the current approach, just added a remark in the test file; if they still want to use them separately they can inject both; but you're not forced to.